ambry-cloud/src/main/java/com/github/ambry/cloud/CloudStorageCompactor.java

```diff
@@ -57,141 +51,130 @@ public CloudStorageCompactor(CloudDestination cloudDestination, CloudConfig clou
     this.cloudDestination = cloudDestination;
     this.partitions = partitions;
     this.vcrMetrics = vcrMetrics;
-    this.shutDownTimeoutSecs = cloudConfig.cloudBlobCompactionShutdownTimeoutSecs;
-    this.numThreads = cloudConfig.cloudCompactionNumThreads;
-    compactionTimeLimitMs = TimeUnit.HOURS.toMillis(cloudConfig.cloudBlobCompactionIntervalHours);
-    logger.info("Allocating {} threads for compaction", numThreads);
-    executorService = Executors.newFixedThreadPool(numThreads);
-    executorCompletionService = new ExecutorCompletionService<>(executorService);
-    doneLatch.set(new CountDownLatch(0));
+    this.cloudConfig = cloudConfig;
+    // Give threads a name, so they can be identified in a thread-dump and set them as daemon or background
+    this.executorService = Utils.newScheduler(this.cloudConfig.cloudCompactionNumThreads, "cloud-compaction-worker-", true);
```

Comment:
Are you sure we need to set the thread type to daemon? It seems like they are used for tasks such as gc that don't prevent jvm from shutting down. In our case, if ambry-vcr main thread is shutting down, we may want to shutdown all the background threads running in it. I think we use non-daemon threads in our code everywhere.

Reply:
All ambry server threads will cease to exist once the jvm exits - daemon or not. Daemons offer the flexibility to exit without requiring an explicit shutdown. Compaction workers are daemons, just issuing deletes to Azure and will not cause any data inconsistencies. However, the code is still offering a chance at a graceful shutdown. In spite of that if the daemons do not exit, then we forcibly shutdown everything.